### PR TITLE
Workflows for Featuretools & Woodwork main

### PR DESCRIPTION
**Pull request by chukarsten**
Fixes: #3478, #3479

### CODE REVIEW

**Comment by chukarsten** [comment]
Was this what you were looking for @gsheni ?

File: .github/workflows/unit_tests_with_featuretools_main_branch.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        libraries: ["core"]
+    steps:
+      - name: Set up python ${{ matrix.python_version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python_version }}
+      - name: Checkout repository
+        uses: actions/checkout@v2
+      - name: Build featuretools package
+        run: make package_featuretools
```
[the remark this answers is not in the document]
Comment (gsheni) [reply .github/workflows/unit_tests_with_featuretools_main_branch.yml:22]: You will need to add command to Makefile
```
.PHONY: package_evalml
package_evalml:
	python setup.py sdist
	$(eval EM_VERSION=$(shell python setup.py --version))
	tar -zxvf "dist/evalml-${EM_VERSION}.tar.gz"
	mv "evalml-${EM_VERSION}" unpacked_sdist
```

File: .github/workflows/unit_tests_with_featuretools_main_branch.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+      - name: Set up pip and graphviz
+        run: |
+          pip config --site set global.progress_bar off
+          python -m pip install -U pip
+          sudo apt update && sudo apt install -y graphviz
+      - name: Install EvalML & Featuretools - test requirements
+        run: |
+          python -m pip install -e ./
```
Comment (gsheni) [inline .github/workflows/unit_tests_with_featuretools_main_branch.yml:30]:
```suggestion
          python -m pip install -e unpacked_sdist/[test]
```

File: .github/workflows/unit_tests_with_featuretools_main_branch.yml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        if: ${{ needs.unit_tests_featuretools_main.result != 'success' }}
+        id: slack
+        uses: slackapi/slack-github-action@v1
+        with:
+          payload: |
+            {
+              "url": "${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}"
+            }
+        env:
+          SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL }}
```
Comment (gsheni) [inline .github/workflows/unit_tests_with_featuretools_main_branch.yml:54]: You will need to setup a Slack webhook
- https://github.com/slackapi/slack-github-action#setup
I added you as a collab on the Featuretools one so you can see what it looks like

File: .github/workflows/unit_tests_with_woodwork_main_branch.yml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python_version }}
+      - name: Checkout repository
+        uses: actions/checkout@v2
+      - name: Set up pip and graphviz
+        run: |
+          pip config --site set global.progress_bar off
+          python -m pip install -U pip
+          sudo apt update && sudo apt install -y graphviz
```
Comment (gsheni) [inline .github/workflows/unit_tests_with_woodwork_main_branch.yml:25]: same, do you need graphviz

File: .github/workflows/unit_tests_with_woodwork_main_branch.yml [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+      - name: Install EvalML & Woodwork - test requirements
+        run: |
+          python -m pip install -e ./
```
Comment (gsheni) [inline .github/workflows/unit_tests_with_woodwork_main_branch.yml:28]:
```suggestion
          python -m pip install -e unpacked_sdist/[test]
```

File: .github/workflows/unit_tests_with_woodwork_main_branch.yml [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+        if: ${{ needs.unit_tests_woodwork_main.result != 'success' }}
+        id: slack
+        uses: slackapi/slack-github-action@v1
+        with:
+          payload: |
+            {
+              "url": "${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}"
+            }
+        env:
+          SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL }}
```
Comment (gsheni) [inline .github/workflows/unit_tests_with_woodwork_main_branch.yml:52]: You will need to setup another slack webhook

File: .github/workflows/unit_tests_with_featuretools_main_branch.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (chukarsten) [reply .github/workflows/unit_tests_with_featuretools_main_branch.yml:22]: Good stuff, thanks, didn't understand exactly what was happening here. I thought it was something featuretools specific

File: .github/workflows/unit_tests_with_featuretools_main_branch.yml [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+          python-version: ${{ matrix.python_version }}
+      - name: Checkout repository
+        uses: actions/checkout@v2
+      - name: Build featuretools package
+        run: make package_featuretools
+      - name: Set up pip and graphviz
+        run: |
+          pip config --site set global.progress_bar off
+          python -m pip install -U pip
+          sudo apt update && sudo apt install -y graphviz
```
[the remark this answers is not in the document]
Comment (chukarsten) [reply .github/workflows/unit_tests_with_featuretools_main_branch.yml:27]: Nope!

File: .github/workflows/unit_tests_with_woodwork_main_branch.yml (hunk 6 above)
Comment (gsheni) [reply .github/workflows/unit_tests_with_woodwork_main_branch.yml:51]: FYI, there will need to be 2 Slack Webhook URLs

File: .github/workflows/unit_tests_with_woodwork_main_branch.yml [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+      - name: Install EvalML & Woodwork - test requirements
+        run: |
+          python -m pip install -e unpacked_sdist/[test]
+          python -m pip uninstall -y woodwork
+          python -m pip install https://github.com/alteryx/woodwork/archive/main.zip
+      - name: Run unit tests without code coverage
+        run: |
+          echo "Run unit tests without code coverage for ${{ matrix.python_version }} and ${{ matrix.libraries }}"
+          echo "Testing with EvalML version:" `python -c "import evalml; print(evalml.__version__)"`
+          pytest evalml/ -n 2
```
Comment (gsheni) [inline .github/workflows/unit_tests_with_woodwork_main_branch.yml:34]:
```suggestion
          pytest evalml/ -n 2
          pip check
```

File: .github/workflows/unit_tests_with_featuretools_main_branch.yml [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+        if: ${{ needs.unit_tests_featuretools_main.result != 'success' }}
+        id: slack
+        uses: slackapi/slack-github-action@v1
+        with:
+          payload: |
+            {
+              "url": "${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}"
+            }
+        env:
+          SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL }}
```
Comment (gsheni) [inline .github/workflows/unit_tests_with_featuretools_main_branch.yml:54]: This secret name needs to be different than the other.

**Review by gsheni — APPROVED** [review]

File: .github/workflows/unit_tests_with_featuretools_main_branch.yml [hunk 10]
```diff
@@ -0,0 +1,54 @@
+name: Unit Tests - EvalML with Featuretools main branch
+on:
```
Comment (freddyaboulton) [inline .github/workflows/unit_tests_with_featuretools_main_branch.yml:2]: Sorry I'm late to this PR! Awesome job @chukarsten and thanks for reviewing @gsheni ! 

Don't we need to add a workflow to woodwork and featuretools to kick off this job on commits to main?

Similar to this: https://github.com/alteryx/woodwork/blob/main/.github/workflows/kickoff_featuretools_unit_tests.yml

File: .github/workflows/unit_tests_with_woodwork_main_branch.yml [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+        python_version: ["3.7", "3.8", "3.9", "3.10"]
+        libraries: ["core"]
+    steps:
+      - name: Set up python ${{ matrix.python_version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python_version }}
+      - name: Checkout repository
+        uses: actions/checkout@v2
+      - name: Set up pip and graphviz
```
Comment (freddyaboulton) [inline .github/workflows/unit_tests_with_woodwork_main_branch.yml:21]: @chukarsten I think we forgot the `make package_evalml` step here

File: .github/workflows/unit_tests_with_featuretools_main_branch.yml (hunk 10 above)
Comment (gsheni) [reply .github/workflows/unit_tests_with_featuretools_main_branch.yml:2]: Added:
https://github.com/alteryx/featuretools/pull/2072

**Pull request merged**